pygen/pygen_src/riscv_directed_instr_lib.py

```diff
@@ -161,3 +163,134 @@ def post_randomize(self):
             instr = self.randomize_gpr(instr)
             self.instr_list.append(instr)
         super().post_randomize()
+
+
+# Push Stack Instructions
+class riscv_push_stack_instr(riscv_rand_instr_stream):
+
+    def __init__(self):
+        super().__init__()
+        self.stack_len = 0
+        self.num_of_reg_to_save = 0
+        self.num_of_redudant_instr = 0
+        self.push_stack_instr = []
+        self.saved_regs = []
+        self.branch_instr = riscv_instr()
+        self.enable_branch = vsc.rand_bit_t()
+        self.push_start_label = ''
+
+    def init(self):
+        self.reserved_rd = [cfg.ra]
+        self.saved_regs = [cfg.ra]
+        self.num_of_reg_to_save = len(self.saved_regs)
+        if self.num_of_reg_to_save * (rcs.XLEN / 8) > self.stack_len:
+            logging.error('stack len [%0d] is not enough to store %d regs',
```

Comment:
should there be a `sys.exit(...)` call here? I think this error case warrants an exit from the generation, wdyt?

Reply:
Yes, Agreed!